src/realm/util/file.cpp

```diff
@@ -1236,10 +1237,25 @@ const char* File::get_encryption_key()
 }
 
 
-#ifndef _WIN32
-
 DirScanner::DirScanner(const std::string& path, bool allow_missing)
 {
+#ifdef _WIN32
+    WIN32_FIND_DATA file;
+    HANDLE search_handle = FindFirstFile((path + "\\*").c_str(), &file);
+    if (search_handle != INVALID_HANDLE_VALUE)
+    {
+        do
+        {
+            if(std::string(file.cFileName) != "." && std::string(file.cFileName) != "..") {
```

Comment:
Can’t we use something like `strcmp` here instead of allocationg `std::string`s?